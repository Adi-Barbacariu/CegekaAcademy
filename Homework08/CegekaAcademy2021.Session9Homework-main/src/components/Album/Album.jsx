import React from "react";
import { Card, Button, Icon, Label, Divider } from "semantic-ui-react";
import "./Album.css";
import WithLightbox from "../Common/WithLightbox.js";
import { DeleteButton, EditAlbumButton } from "../../components/Common";

const SlideshowButton = ({ photos }) => (
  <Button
    attached="bottom"
    fluid
    basic
    color="blue"
    style={{ margin: "5px 0" }}
  >
    <WithLightbox photos={photos}>Slideshow</WithLightbox>
  </Button>
);

function Album({
  index,
  name,
  description,
  tags,
  photos,
  deleteAlbum,
  editAlbum,
}) {
  const renderTags = (tagsArray) =>
    tagsArray.map((t, idx) => <Label key={idx}>{t}</Label>);

  return (
    <Card>
      <Card.Content>
        <Card.Header className="header">{name}</Card.Header>
        <Divider />
        {renderTags(tags)}
        <Card.Description className="album-desc">
          {description}
        </Card.Description>
      </Card.Content>

      <Card.Content extra>
        <SlideshowButton photos={photos} />
        <div className="ui two buttons">
          {/* <Button basic color="green">
            <Icon name="edit" />
            Edit
          </Button> */}
          <EditAlbumButton
            index={index}
            name={name}
            description={description}
            tags={tags}
            photos={photos}
            editAlbum={editAlbum}
          />
          <DeleteButton
            objectName={name}
            index={index}
            deleteObject={(e) => {
              deleteAlbum(e, index);
            }}
          />
        </div>
      </Card.Content>
    </Card>
  );
}

export default Album;
