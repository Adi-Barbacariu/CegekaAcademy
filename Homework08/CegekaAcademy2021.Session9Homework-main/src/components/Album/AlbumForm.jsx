import React from "react";

function AlbumForm() {
  return <div>album form</div>;
}

export default AlbumForm;
