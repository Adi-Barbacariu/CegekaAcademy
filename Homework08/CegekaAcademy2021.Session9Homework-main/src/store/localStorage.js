import * as api from "../api";

export const loadPhotos = () => {
  const localPhotos = localStorage.getItem("photos");

  if (localPhotos) {
    return JSON.parse(localPhotos);
  }

  return api.getPhotos();
};

export const loadAlbums = () => {
  const localAlbums = localStorage.getItem("albums");

  if (localAlbums) {
    return JSON.parse(localAlbums);
  }

  return api.getAlbums();
};

export const saveState = (itemName, state) => {
  localStorage.setItem(itemName, JSON.stringify(state));
};
