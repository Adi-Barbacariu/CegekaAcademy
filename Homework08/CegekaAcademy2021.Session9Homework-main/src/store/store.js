import { configureStore } from "@reduxjs/toolkit";
import photosReducer from "../features/photosSlice";
import albumsReducer from "../features/albumsSlice";

export default configureStore({
  reducer: {
    photos: photosReducer,
    albums: albumsReducer,
  },
});
