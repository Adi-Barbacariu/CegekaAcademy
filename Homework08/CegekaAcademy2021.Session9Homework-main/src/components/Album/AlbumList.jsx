import React from "react";
import Album from "./Album";
import { Card } from "semantic-ui-react";
import StatusBar from "../StatusBar/StatusBar";

const getAlbumPhotos = (photos, photosIds) => {
  const albumPhotos = [];

  photosIds.forEach((id) => {
    for (let [key, value] of Object.entries(photos)) {
      if (key == id) albumPhotos.push(value);
    }
  });

  return albumPhotos;
};

function AlbumList({ albums, photos }) {
  const albumCount = Object.keys(albums).length;

  return (
    <React.Fragment>
      <StatusBar itemType="album" itemCount={albumCount} photos={photos} />
      <Card.Group>
        {Object.keys(albums).map((key) => {
          const currentAlbum = albums[key];

          const albumPhotos = getAlbumPhotos(photos, currentAlbum.photosIds);

          return (
            <Album
              index={key}
              {...currentAlbum}
              photos={albumPhotos}
              key={key}
            />
          );
        })}
      </Card.Group>
    </React.Fragment>
  );
}

export default AlbumList;
