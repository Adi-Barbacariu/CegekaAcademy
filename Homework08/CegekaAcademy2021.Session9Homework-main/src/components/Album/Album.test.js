import React from "react";
import { Provider } from "react-redux";
import store from "../../store/store";
import Album from "../Album/Album.jsx";

describe("Album", () => {
  it("Album snapshot test", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Album
            index="key"
            name="album name"
            description="album description"
            tags={["tag1", "tag2"]}
            photos={["photo1", "photo2"]}
            deleteAlbum="delete album"
            editAlbum="edit album"
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
