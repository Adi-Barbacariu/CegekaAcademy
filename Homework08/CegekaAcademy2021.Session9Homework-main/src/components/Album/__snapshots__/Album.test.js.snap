// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album Album snapshot test 1`] = `
<div
  className="ui card"
  href={undefined}
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header header"
    >
      album name
    </div>
    <div
      className="ui divider"
    />
    <div
      className="ui label"
      onClick={[Function]}
    >
      tag1
    </div>
    <div
      className="ui label"
      onClick={[Function]}
    >
      tag2
    </div>
    <div
      className="description album-desc"
    >
      album description
    </div>
  </div>
  <div
    className="extra content"
  >
    <div
      className="ui blue basic fluid bottom attached button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      style={
        Object {
          "margin": "5px 0",
        }
      }
      tabIndex={0}
    >
      <div
        onClick={[Function]}
      >
        <div>
          Slideshow
        </div>
      </div>
    </div>
    <div
      className="ui two buttons"
    >
      <button
        className="ui green basic button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="edit icon"
        />
        Edit
      </button>
      <button
        className="ui red basic button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="trash icon"
        />
        Delete
      </button>
    </div>
  </div>
</div>
`;
